sequence_info.py

```diff
@@ -0,0 +1,115 @@
+import flywheel
+
+seqinfo_fields = [
+    'total_files_till_now',  # 0
+    'example_dcm_file',      # 1
+    'series_id',             # 2
+    'dcm_dir_name',          # 3
+    'unspecified2',          # 4
+    'unspecified3',          # 5
+    'dim1', 'dim2', 'dim3', 'dim4', # 6, 7, 8, 9
+    'TR', 'TE',              # 10, 11
+    'protocol_name',         # 12
+    'is_motion_corrected',   # 13
+    'is_derived',            # 14
+    'patient_id',            # 15
+    'study_description',     # 16
+    'referring_physician_name', # 17
+    'series_description',    # 18
+    'sequence_name',         # 19
+    'image_type',            # 20
+    'accession_number',      # 21
+    'patient_age',           # 22
+    'patient_sex',           # 23
+    'date',                  # 24
+    'series_uid'             # 25
+ ]
+
+SeqInfo = namedtuple('SeqInfo', seqinfo_fields)
+
+
+def seqinfo_from_acquisition(acquisition):
+    """Using a flywheel acquisition object and the files in it, generates a
+    "SeqInfo" object that is passed to the heuristics
+
+    NOTE: The acquisition object must be a full object, not one returned from
+    a list or finder, as it requires the info fields of itself and its files
+
+    Args:
+        acquisition (Acquisition): A flywheel acquisition object
+
+    Returns:
+        SeqInfo: A named tuple containing the field defined in swqinfo_fields
+        None: acquisition doesn't have a dicom
+    """
+    dicoms = [ f for f in acquisition.files if f.type == 'dicom' ]
+    if dicoms:
+        dicom = dicoms[0]
+    else:
+        return None
+
+
+    # Grab a single image for the example
+    zip_info = acquisition.get_file_zip_info(dicom.name)
+    example_dcm_file = zip_info.members[0].path
+
+    total = len(zip_info.members)
+
+    series_id = dicom.info.get('SeriesId')
+    dcm_dir_name = dicom.name
+    # # Not sure what the first 3 dimension refer to, I didn't see
+    # sizes = len(zip_info.members)
+    TR = float(dicom.info.get('RepititionTime', -1000)/1000.
```

Comment:
Is it guaranteed that these will be in seconds? I've seen some in milliseconds. Maybe we should add a check here or is this the wrong place?

Reply:
Looking at the Dicom Standard it looks like milliseconds is the required format for the value, which confuses me [here](https://github.com/nipy/heudiconv/blob/master/heudiconv/dicoms.py#L174) in the heudiconv tool where the conversion is happening

Reply:
Could be that one of the upstream dicom readers is converting them to seconds? Maybe this isn't worth looking into too deeply at the moment. As long as this is consistent within a project the heuristic can be changed to match the units here.

Reply:
Also, I'll start merging some of the work here to the work done in `query.py`.

As a side note, for the dcm_dir_name does it make sense to use the dicom file name, or the acquisition label? For the upload of the bbl data from xnat for example the acquisition labels are the names of the directories the dicom series were in.

Reply:
I'd be surprised if that is ever actually used in a heuristic. We could set this to the name of the dicom zip file, otherwise re-use the series_uid value.